client/payment-request/index.js

```diff
@@ -388,8 +388,10 @@ jQuery( ( $ ) => {
 						).then( () => {
 							wcpayPaymentRequest.unblockPaymentRequestButton();
 						} );
-					}
-				);
+					} )
+					.catch( () => {
+						wcpayPaymentRequest.hide();
```

Comment:
If the server throws an exception, or returns an error after a caught one (as will be the case here), this means that we don't have enough data to display the overlay, which is why it's getting hidden here.

We need to decide if this is good UX, considering that the standard payment flow is still available, or design feedback would be required.

FWIW, this is a real edge case, for which I cannot find steps to reproduce. As indicated in the testing instructions of the PR, we need to intermittently change PHP code, in order to even test it.

Reply:
I feel the error in this case from the server should have been a `400` instead of a `500` and should have been handled accordingly here. The request is for an intermittent state which will be handled on refresh by that option not showing anymore so a `500` doesn't seem apt but not a blocker.